core-js/src/main/javascript/cdf/dashboard/Dashboard.dataSources.js

```diff
@@ -204,6 +204,17 @@ define([
       } else {
         Logger.warn("Data source name '" + name + "' not found");
       }
+    },
+
+    /**
+     * @summary Clears resources associated with the dashboard instance datasources.
+     * @description  Dispose resources that the dashboard datasources may have and that are no longer needed.
+     *
+     */
+    disposeDataSources: function() {
+      if (this.dataSources && this.dataSources.dataSource && this.dataSources.dataSource) {
```

Comment:
Duplicated condition - this.dataSources.dataSource

Reply:
This method should be marked protected (and be named `_disposeDataSources`).